icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -0,0 +1,366 @@
+                measures5.get(1).getNumber());
+        assertEquals("light-year test: measures[1] unit", MeasureUnit.METER.getIdentifier(),
+                measures5.get(1).getUnit().getIdentifier());
+
+        // TODO(icu-units#63): test negative numbers!
+    }
+
+
+    @Test
+    public void testComplexUnitConverterSorting() {
+
+        MeasureUnitImpl source = MeasureUnitImpl.forIdentifier("meter");
+        MeasureUnitImpl target = MeasureUnitImpl.forIdentifier("inch-and-foot");
+        ConversionRates conversionRates = new ConversionRates();
+
+        ComplexUnitsConverter complexConverter = new ComplexUnitsConverter(source, target, conversionRates);
+        List<Measure> measures = complexConverter.convert(BigDecimal.valueOf(10.0));
+
+        assertEquals(measures.size(), 2);
+        assertEquals("inch-and-foot unit 0", "foot", measures.get(0).getUnit().getIdentifier());
+        assertEquals("inch-and-foot unit 1", "inch", measures.get(1).getUnit().getIdentifier());
+
+        assertTrue("inch-and-foot value 0", compareTwoBigDecimal(BigDecimal.valueOf(32), BigDecimal.valueOf(measures.get(0).getNumber().doubleValue()), BigDecimal.valueOf(0.0001)));
+        assertTrue("inch-and-foot value 1", compareTwoBigDecimal(BigDecimal.valueOf(9.7008), BigDecimal.valueOf(measures.get(1).getNumber().doubleValue()), BigDecimal.valueOf(0.0001)));
+    }
+
+
+    @Test
+    public void testExtractConvertibility() {
+        class TestData {
+            MeasureUnitImpl source;
+            MeasureUnitImpl target;
+            UnitConverter.Convertibility expected;
+
+            TestData(String source, String target, UnitConverter.Convertibility convertibility) {
+                this.source = MeasureUnitImpl.UnitsParser.parseForIdentifier(source);
+                this.target = MeasureUnitImpl.UnitsParser.parseForIdentifier(target);
+                this.expected = convertibility;
+            }
+        }
+
+        TestData[] tests = {
+                new TestData("meter", "foot", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("square-meter-per-square-hour", "hectare-per-square-second", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("hertz", "revolution-per-second", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("millimeter", "meter", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("yard", "meter", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("ounce-troy", "kilogram", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("percent", "portion", UnitConverter.Convertibility.CONVERTIBLE),
+                new TestData("ofhg", "kilogram-per-square-meter-square-second", UnitConverter.Convertibility.CONVERTIBLE),
+
+                new TestData("second-per-meter", "meter-per-second", UnitConverter.Convertibility.RECIPROCAL),
+        };
```

Comment:
Please add these test cases:

        {"kilometer", "foot", CONVERTIBLE},                                          //
        {"hectare", "square-foot", CONVERTIBLE},                                     //
        {"kilometer-per-second", "second-per-meter", RECIPROCAL},                    //
        {"square-meter", "square-foot", CONVERTIBLE},                                //
        {"kilometer-per-second", "foot-per-second", CONVERTIBLE},                    //
        {"square-hectare", "pow4-foot", CONVERTIBLE},                                //
        {"square-kilometer-per-second", "second-per-square-meter", RECIPROCAL},      //
        {"cubic-kilometer-per-second-meter", "second-per-square-meter", RECIPROCAL}, //

I will attempt to make ICU4J and ICU4C unit testing much more data-driven in the future (so we can share tests more easily).

Reply:
this will be synced later

Reply:
Do we have a ticket or TODO to track this syncing?

Reply:
I have taken care of this in https://github.com/unicode-org/icu/pull/1421